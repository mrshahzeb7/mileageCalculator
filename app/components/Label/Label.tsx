import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {colors, spacing} from '../../theme';

interface Props {
  label: string | undefined;
}

export const Label: React.FC<Props> = ({label}) => {
  return !!label && <Text style={styles.labelStyle}>{label}</Text>;
};

// styles
const styles = StyleSheet.create({
  labelStyle: {
    marginLeft: spacing.horizontal.medium,
    backgroundColor: colors.primary,
    paddingHorizontal: spacing.horizontal.tiny,
    zIndex: 1,
    color: colors.textDim,
    alignSelf: 'flex-start',
    position: 'absolute',
    top: -spacing.vertical.tiny,
  },
});
