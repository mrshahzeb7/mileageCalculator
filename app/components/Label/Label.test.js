import 'react-native';
import React from 'react';
import {Label} from './Label';
import renderer from 'react-test-renderer';

describe('Label', () => {
  it(' Label renders correctly', () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
