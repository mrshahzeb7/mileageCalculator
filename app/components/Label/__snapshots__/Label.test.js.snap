// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label  Label renders correctly 1`] = `null`;
